[ADD] repairs: add repair orders doc (#8647)
odoo/documentation: content/applications/inventory_and_mrp/repairs/repair_orders.rst

=====================
Process repair orders
=====================

.. |SO| replace:: :abbr:`SO (Sales Order)`
.. |DO| replace:: :abbr:`DO (Delivery Order)`
.. |RO| replace:: :abbr:`RO (Repair Order)`
.. |UoM| replace:: :abbr:`UoM (Unit of Measure)`

Sometimes, products delivered to customers can break or be damaged in transit, and need to be
returned for a refund, delivery of a replacement product, or repairs.

In Odoo, repairs for products returned by customers can be tracked in the *Repairs* app. Once

> repaired, products can be redelivered to the customer.

Reviewer: Since this is a complicated workflow with many steps, how about outlining the steps, with links to relevant sections at the top here before we jump in? It could give the reader a big picture summary so they can keep track of all the steps. For example,

1. Sell and deliver product
2. Customer returns damaged product (ref link to "Return order" section of your doc)
3. Create and process RO (link to "Create Repair Order" section of your doc)
4. Re-deliver product to customer (link to "Return product to customer")

Author: Added ✅

Return order
============

> Returns can be processed in Odoo via *Reverse Transfers*, created directly from a sales order (SO)

Author: Suggestion implemented ✅

   .. image:: repair_orders/repair-orders-quantity-delivered.png
      :align: center
      :alt: Delivered and Quantity columns on sales order after return.

Create repair order
===================

Once products have been returned, their repairs can be tracked by creating a repair order (RO).

To create a new |RO|, navigate to :menuselection:`Repairs app`, and click :guilabel:`New`. This
opens a blank |RO| form.

.. image:: repair_orders/repair-orders-left-hand-form.png
   :align: center
   :alt: Left-hand side of blank repair order form.

On this form, begin by selecting a :guilabel:`Customer`. The customer selected should be for whom
the order will be invoiced and delivered.

In the :guilabel:`Product to Repair` field, click the drop-down menu to select the product that
needs repair. If necessary, click :guilabel:`Search More...` to open a :guilabel:`Search: Product to
Repair` pop-up window, and browse all products in the database.

Once a :guilabel:`Product to Repair` is selected, a new :guilabel:`Product Quantity` field appears
below it. In that field, enter the quantity (in a `0.00` format) of the product that requires
repair.

> To the right of that value, click the drop-down list to select the Unit of Measure (UoM) for the

Author: Suggestion implemented ✅

product.

In the :guilabel:`Return` field, click the drop-down menu and select the return order from which the

> product to be repaired comes from.

Author: It looks like the product you used for the return was [FURN_8855] Drawer, and on the repair you listed a different product [FURN_2100] Drawer Black. That's why it didn't show up

> product to be repaired comes from.

Reviewer: oh my goodness *face palm*. Thank you for clarifying. I'll try it again!

> product to be repaired comes from.

Reviewer: Tried again with the correct product selected, and it works as expected! Thank you for clarifying for me!

In the :guilabel:`Under Warranty` field, tick the checkbox if the product being repaired is covered
by a warranty. If ticked, the sales price is set to `0` for all products transferred from the repair
order.

In the :guilabel:`Scheduled Date` field, click the date to reveal a calendar popover window. From
this calendar, select a date for the repair, and click :guilabel:`Apply`.

.. image:: repair_orders/repair-orders-right-hand-form.png
   :align: center
   :alt: Right-hand side of blank repair order form.

In the :guilabel:`Responsible` field, click the drop-down menu and select the user who should be
responsible for the repair.

In the :guilabel:`Company` field, if in a multi-company environment, select which company this |RO|
belongs to.

In the :guilabel:`Tags` field, click the drop-down menu and select which tags should be applied to
this |RO|.

Parts tab
---------

With the :guilabel:`Parts` tab selected, at the bottom of the form, click :guilabel:`Add a line`.

> Doing so creates a new line, where parts can be added, removed, or recycled.

Author: Suggestion implemented ✅

:guilabel:`Add` (selected by default), :guilabel:`Remove`, and :guilabel:`Recycle`.

.. image:: repair_orders/repair-orders-type-column.png
   :align: center
   :alt: Type column options or new part under Parts tab.

Choosing :guilabel:`Add` adds this part to the |RO|.

Choosing :guilabel:`Remove` removes this part from the |RO|.

Choosing :guilabel:`Recycle` recycles this part from the |RO|, designating it for later use or to be
repurposed for another use in the warehouse.

In the :guilabel:`Product` column, select which product (part) should be added, removed, or
recycled. In the :guilabel:`Demand` column, change the quantity, if necessary, to indicate what
quantity of this part should be used in the repair process.

In the :guilabel:`Done` column, change the value (in a `0.00` format) once the part has been
successfully added, removed, or recycled.

In the :guilabel:`Unit of Measure` column, select the |UoM| for the part.

Finally, in the :guilabel:`Used` column, tick the checkbox once the part has been used in the repair
process.

To add additional columns to the line, click the :guilabel:`optional columns drop-down toggle (two

> dots)` icon, at the far-right of the header row. Select the desired options to add to the line.

Author: I chose the term I did because it's what it read when I inspected the code, but now that I know it should be slider icon, I'll use that from now on. Suggestion implemented ✅

.. image:: repair_orders/repair-orders-additional-options.png
   :align: center
   :alt: Optional additional options to add to new part line.

Repair Notes and Miscellaneous tabs
-----------------------------------

Click the :guilabel:`Repair Notes` tab to add internal notes about this specific |RO|, and anything
the user performing the repair might need to know.

> Click the blank space (with the greyed-out :guilabel:`Add internal notes.` instruction) to begin

Author: Suggestion implemented ✅

writing notes.

Click the :guilabel:`Miscellaneous` tab to see the :guilabel:`Operation Type` for this repair. By
default, this is set to :guilabel:`YourCompany: Repairs`, indicating this is a repair type
operation.

Once all desired configurations have been made on the |RO| form, click :guilabel:`Confirm Repair`.
This moves the |RO| to the :guilabel:`Confirmed` stage, and reserves the necessary components needed
for the repair.

A new :guilabel:`Forecasted` column appears on the product lines under the :guilabel:`Parts` tab,
displaying the availability of all components needed for the repair.

Once ready, click :guilabel:`Start Repair`. This moves the |RO| to the :guilabel:`Under Repair`

> stage. If the |RO| should be canceled, click :guilabel:`Cancel Repair`.

Author: Suggestion implemented ✅

Once all products have been successfully repaired, the |RO| is completed. To register this in the
database, click :guilabel:`End Repair`.

.. note::
   If all parts added to the |RO| were not used, clicking :guilabel:`End Repair` causes an
   :guilabel:`Uncomplete Move(s)` pop-up window to appear.

   .. image:: repair_orders/repair-orders-uncomplete-moves.png
      :align: center
      :alt: Uncomplete Moves pop-up window for unused parts.

   The pop-up window informs the user that there is a difference between the initial demand and the
   actual quantity used for the order.

   If the :guilabel:`Used` quantity should be changed, click :guilabel:`Discard` or :guilabel:`X`
   out of the pop-up window to make the change. If the order should be confirmed, click

>    :guilabel:`Validate`.

Author: Suggestion implemented ✅

This moves the |RO| to the :guilabel:`Repaired` stage. A :guilabel:`Product Moves` smart button also
appears above the form.

Click the :guilabel:`Product Moves` smart button to view the product's moves history before, during,

> and after the repair process.

Author: Nice catch, small detail I overlooked with the wording I chose. Changed it accordingly. Suggestion implemented ✅

.. image:: repair_orders/repair-orders-product-moves.png
   :align: center
   :alt: Moves history of product included in the repair order.

Return product to customer
--------------------------

Once the product has been successfully repaired, it can be returned to the customer.

> If the product is not covered by a warranty, or the customer should be charged for repairs, click

Author: Suggestion implemented ✅

Once products have been returned, their repairs can be tracked by creating a repair order (RO).

To create a new |RO|, navigate to :menuselection:`Repairs app`, and click :guilabel:`New`. This
opens a blank |RO| form.

.. image:: repair_orders/repair-orders-left-hand-form.png
   :align: center
   :alt: Left-hand side of blank repair order form.

On this form, begin by selecting a :guilabel:`Customer`. The customer selected should be for whom
the order will be invoiced and delivered.

In the :guilabel:`Product to Repair` field, click the drop-down menu to select the product that
needs repair. If necessary, click :guilabel:`Search More...` to open a :guilabel:`Search: Product to
Repair` pop-up window, and browse all products in the database.

Once a :guilabel:`Product to Repair` is selected, a new :guilabel:`Product Quantity` field appears
below it. In that field, enter the quantity (in a `0.00` format) of the product that requires
repair.

To the right of that value, click the drop-down list to select the unit of measure (UoM) for the
product.

In the :guilabel:`Return` field, click the drop-down menu and select the return order from which the
product to be repaired comes from.

> In the :guilabel:`Under Warranty` field, tick the checkbox if the product being repaired is covered
> by a warranty. If ticked, the sales price is set to `0` for all products transferred from the repair
> order.

Reviewer: Totally up to you how you'd like to describe this, but here's a suggestion!
```suggestion
Tick the :guilabel:`Under Warranty` checkbox, if the product being repaired is covered
by a warranty. If ticked, the :guilabel:`Customer` is not charged for all the parts used in the repair
order.
```

Is this still correct? I wanted to clarify your original sentence:
> If ticked, the sales price is set to `0` for all products transferred from the repair
order.

Author: Suggestion implemented ✅

In the :guilabel:`Scheduled Date` field, click the date to reveal a calendar popover window. From
this calendar, select a date for the repair, and click :guilabel:`Apply`.

> .. image:: repair_orders/repair-orders-right-hand-form.png

Reviewer 2: Optional:
I don't think this screenshot  is as informative as a fully completed RO form, to visually demonstrate for the reader what the written context is describing. Would consider switching 😉

Author: Replaced screenshot with zoomed-out RO image, showing everything filled out. Suggestion implemented ✅

   :align: center
   :alt: Right-hand side of blank repair order form.

In the :guilabel:`Responsible` field, click the drop-down menu and select the user who should be
responsible for the repair.

In the :guilabel:`Company` field, if in a multi-company environment, select which company this |RO|
belongs to.

In the :guilabel:`Tags` field, click the drop-down menu and select which tags should be applied to
this |RO|.

Parts tab
---------

Add, remove, or recycle parts in the :guilabel:`Parts` tab. To do so, click :guilabel:`Add a line`
at the bottom of the form.

In the :guilabel:`Type` column, click the box to reveal three options to choose from:
:guilabel:`Add` (selected by default), :guilabel:`Remove`, and :guilabel:`Recycle`.

.. image:: repair_orders/repair-orders-type-column.png
   :align: center
   :alt: Type column options or new part under Parts tab.

Choosing :guilabel:`Add` adds this part to the |RO|.

Choosing :guilabel:`Remove` removes this part from the |RO|.

Choosing :guilabel:`Recycle` recycles this part from the |RO|, designating it for later use or to be

> repurposed for another use in the warehouse.

Reviewer 2: It's clear to me that this is what these options are for— the bigger question I have is what exactly are we doing on the _Parts_ tab? What's it for? For what purpose would we Add, Remove, or Recycle parts here?

stage (in the upper-right corner). If the |RO| should be canceled, click :guilabel:`Cancel Repair`.

Once all products have been successfully repaired, the |RO| is completed. To register this in the
database, click :guilabel:`End Repair`.

.. note::
   If all parts added to the |RO| were not used, clicking :guilabel:`End Repair` causes an
   :guilabel:`Uncomplete Move(s)` pop-up window to appear.

   .. image:: repair_orders/repair-orders-uncomplete-moves.png
      :align: center
      :alt: Uncomplete Moves pop-up window for unused parts.

   The pop-up window informs the user that there is a difference between the initial demand and the
   actual quantity used for the order.

   If the :guilabel:`Used` quantity should be changed, click :guilabel:`Discard` or close the pop-up
   window. If the order should be confirmed, click :guilabel:`Validate`.

This moves the |RO| to the :guilabel:`Repaired` stage. A :guilabel:`Product Moves` smart button also
appears above the form.

Click the :guilabel:`Product Moves` smart button to view the product's moves history during and
after the repair process.

> .. image:: repair_orders/repair-orders-product-moves.png

Reviewer 2: Optional: 
I like this shot showing the difference between the locations— consider zooming out to show more context around this section. By itself, it's isolated and isn't clear where exactly we're looking at this in Odoo.

Author: Agreed - replaced screenshot with zoomed-out one showing all relevant areas of the page. Suggestion implemented ✅

   :align: center
   :alt: Moves history of product included in the repair order.

.. _repairs/repair_orders/return-customer:

Return product to customer
--------------------------

Once the product has been successfully repaired, it can be returned to the customer.

> If the product is not under warranty, or should the customer bear the repair costs, click

Reviewer 2: Since the instructions for products not under warranty is completely different then the steps that proceed after this section, consider putting an h4 heading here as well as line 218 to differentiate the two instructions as conditional flows.

Author: Nice catch - added H4 headings here to separate the instances of customer returns. Suggestion implemented ✅

repair. If the components are used, the user completing the repair can record they were used. If
they were not used, the user can indicate that, too, and the components can be saved for another
use.

Choosing :guilabel:`Remove` removes this part from the |RO|. Removing parts lists components that
should be removed from the product being repaired during the repair process. If the parts are
removed, the user completing the repair can indicate they were removed.

Choosing :guilabel:`Recycle` recycles this part from the |RO|, designating it for later use or to be
repurposed for another use in the warehouse.

In the :guilabel:`Product` column, select which product (part) should be added, removed, or
recycled. In the :guilabel:`Demand` column, change the quantity, if necessary, to indicate what
quantity of this part should be used in the repair process.

In the :guilabel:`Done` column, change the value (in a `0.00` format) once the part has been
successfully added, removed, or recycled.

In the :guilabel:`Unit of Measure` column, select the |UoM| for the part.

Finally, in the :guilabel:`Used` column, tick the checkbox once the part has been used in the repair
process.

> To add additional columns to the line, click the :guilabel:`(slider)` icon, at the far-right of the
> header row. Select the desired options to add to the line.

Reviewer 3: Small update to the icon name :)
![image](https://github.com/odoo/documentation/assets/36018073/67f2ceb2-795d-433c-864d-6cbd86c708e1)
```suggestion
To add additional columns to the line, click the :guilabel:`(optional columns drop-down)` icon, at
the far-right of the header row. Select the desired options to add to the line.
```

.. image:: repair_orders/repair-orders-additional-options.png
   :align: center
   :alt: Optional additional options to add to new part line.

Repair Notes and Miscellaneous tabs
-----------------------------------

Click the :guilabel:`Repair Notes` tab to add internal notes about this specific |RO|, and anything
the user performing the repair might need to know.

Click the blank text field to begin writing notes.

Click the :guilabel:`Miscellaneous` tab to see the :guilabel:`Operation Type` for this repair. By

> default, this is set to :guilabel:`YourCompany: Repairs`, indicating this is a repair type

Reviewer 3: Just double checking- should there be a space here?
```suggestion
default, this is set to :guilabel:`Your Company: Repairs`, indicating this is a repair type
```

Author: There's not a space, although I know it looks weird :-/ But that's how it appears in the UI
